homeassistant/components/deconz/deconz_device.py

```diff
@@ -30,20 +30,25 @@ def __init__(
     @property
     def unique_id(self) -> str:
         """Return a unique identifier for this device."""
-        return self._device.unique_id
+        if isinstance(self._device, PydeconzScene):
+            return ""
```

Comment:
These two lines are unreacheable.
Maybe you should move `get_device_identifier` to `DeconzBase` and replace it here with:
```suggestion
        if isinstance(self._device, PydeconzScene):
            return self.get_device_identifier()
```

Then you can also delete the `unique_id` override in `DeconzSceneMixin`

Reply:
The other possibility is to use assert:
```suggestion
        # unique_id is overriden in DeconzSceneMixin
        assert not isinstance(self._device, PydeconzScene)
```

Reply:
I started by rewriting these things but it grew too much in size so I backed down to the simplest way to achieve the strict typing. I will try to continue making small improvements forward to remove bad paths like this.

Is it ok to leave as is for now?

Reply:
I would still use this one, which doesn't introduce dead code, and still makes mypy happy:
```suggestion
        # unique_id is overriden in DeconzSceneMixin
        assert not isinstance(self._device, PydeconzScene)
```

Reply:
Ok!

Reply:
Done